includes/templates/template_default/jscript/zen_addr_pulldowns.php

```diff
@@ -11,58 +11,78 @@
+    //
+    $countries = $db->Execute(
+        "SELECT DISTINCT zone_country_id
            FROM " . TABLE_ZONES . "
-          WHERE zone_country_id = $current_country_id
-       ORDER BY zone_name"
+                INNER JOIN " . TABLE_COUNTRIES . "
+                    ON countries_id = zone_country_id
+                   AND status = 1
+       ORDER BY zone_country_id"
     );
-    foreach ($states as $next_state) {
-        $c2z[$current_country_id][$next_state['zone_id']] = $next_state['zone_name'];
+    foreach ($countries as $next_country) {
+        $current_country_id = $next_country['zone_country_id'];
+        $c2z[$current_country_id] = [];
+
+        $states = zen_get_country_zones($current_country_id);
+        foreach ($states as $next_state) {
+            $c2z[$current_country_id][$next_state['id']] = $next_state['text'];
+        }
     }
 }
-
-if (count($c2z) !== 0) {
+?>
+<script>
+jQuery(document).ready(function() {
+<?php
     echo '    var country_zones = \'' . addslashes(json_encode($c2z)) . '\';' . PHP_EOL;
-}
 
 // -----
-// Initialize the display for the dropdown vs. hand-entry of the state fields.  If the initially-selected
-// country doesn't have zones, the dropdown will contain only 1 element ('Type a choice below ...').
+// Notes:
+//
+// 1. The '#stBreak' <br> is also never needed/wanted since it will 'throw' the state-field input underneath
+// the state/zone label.
+//
+// 2. If the '#stateLabel' label is empty, hide it!  It will be when the site uses dropdown states and on
+// the 'shipping_estimator' page.
+//
+// 3. Initialize the display for the dropdown vs. hand-entry of the state fields.  If the initially-selected
+// country doesn't have zones, the dropdown will contain only 1 element ('Type a choice below ...').  In that
+// case, the dropdown and associated elements will be hidden and the hand-input 'state' field will be shown.
+//
+// 4. There can be unwanted whitespace, e.g. an &nbsp; prior to the (optional) <span class="alert"> following
+// the 'stateZone' dropdown.  In that case, when the <span> is hidden for unzoned countries, the state input
+// field is slightly offset from the other input fields.
 //
 ?>
-    if (jQuery('#stateZone > option').length == 1) {
-        jQuery('#stateZone').hide();
-        jQuery('#state, #stBreak, #stateLabel').show();
-    } else {
-        jQuery('#state, #stBreak, #stateLabel').hide();
+    jQuery('#stBreak').hide();
+    if (jQuery('#stateLabel').text().length == 0) {
+        jQuery('#stateLabel').hide();
+    }
+    if (jQuery('#stateZone > option').length > 1) {
+        jQuery('#state').hide();
         jQuery('#stateZone').show();
+        jQuery('#stateZone').next('span.alert').show();
+    } else {
+        jQuery('#state').show();
+        jQuery('#stateZone').hide();
+        jQuery('#stateZone').next('span.alert').hide();
     }
 
     var pleaseSelect = '<?php echo PLEASE_SELECT; ?>';
```

Comment:
This could be a constant.

Reply:
You mean a script-based one?  How important is that?

Reply:
From what I read, not important in any real sense. Just semantically correct.  
Does it even need to be created at all/could be inline, since it's only used once?

Reply:
Variable now moved inline, using `addslashes` ... just in case to ensure valid javascript created.